# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../paintlib/gcc43.patch
# Copyright (C) 2009 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

diff -Nur paintlib-2.6.2-orig/common/Filter/pl2passscale.h paintlib-2.6.2/common/Filter/pl2passscale.h
--- paintlib-2.6.2-orig/common/Filter/pl2passscale.h	2009-02-09 00:12:04.000000000 +0000
+++ paintlib-2.6.2/common/Filter/pl2passscale.h	2009-02-09 00:16:27.000000000 +0000
@@ -15,6 +15,7 @@
 #include "../plpaintlibdefs.h"
 
 #include <math.h>
+#include <cstring>
 
 // Not every math.h defines this :-(.
 #ifndef  min
diff -Nur paintlib-2.6.2-orig/common/Filter/plfiltercrop.cpp paintlib-2.6.2/common/Filter/plfiltercrop.cpp
--- paintlib-2.6.2-orig/common/Filter/plfiltercrop.cpp	2009-02-09 00:12:04.000000000 +0000
+++ paintlib-2.6.2/common/Filter/plfiltercrop.cpp	2009-02-09 00:14:37.000000000 +0000
@@ -11,7 +11,7 @@
 #include "plfiltercrop.h"
 #include "plbitmap.h"
 #include "plrect.h"
-
+#include <cstring>
 
 PLFilterCrop::PLFilterCrop(int XMin, int YMin, int XMax, int YMax)
   : PLFilter(),
diff -Nur paintlib-2.6.2-orig/common/Filter/plfilterflip.cpp paintlib-2.6.2/common/Filter/plfilterflip.cpp
--- paintlib-2.6.2-orig/common/Filter/plfilterflip.cpp	2009-02-09 00:12:04.000000000 +0000
+++ paintlib-2.6.2/common/Filter/plfilterflip.cpp	2009-02-09 00:22:14.000000000 +0000
@@ -12,6 +12,7 @@
 #include "plbitmap.h"
 #include "plfilterflip.h"
 #include "plpoint.h"
+#include <cstring>
 
 //////////////////////////////////////////////////////////////////////
 // Construction/Destruction
diff -Nur paintlib-2.6.2-orig/common/Filter/plfilterquantize.cpp paintlib-2.6.2/common/Filter/plfilterquantize.cpp
--- paintlib-2.6.2-orig/common/Filter/plfilterquantize.cpp	2009-02-09 00:12:04.000000000 +0000
+++ paintlib-2.6.2/common/Filter/plfilterquantize.cpp	2009-02-09 00:18:27.000000000 +0000
@@ -31,6 +31,7 @@
 #include "plfilterquantize.h"
 #include "plbitmap.h"
 #include "plexcept.h"
+#include <cstring>
 
 #ifndef  min
 #define min(a, b)       ((a) < (b) ? (a) : (b))
diff -Nur paintlib-2.6.2-orig/common/plbitmap.cpp paintlib-2.6.2/common/plbitmap.cpp
--- paintlib-2.6.2-orig/common/plbitmap.cpp	2009-02-09 00:12:04.000000000 +0000
+++ paintlib-2.6.2/common/plbitmap.cpp	2009-02-09 00:28:11.000000000 +0000
@@ -17,6 +17,7 @@
 #include "plfilterquantize.h"
 
 #include <iostream>
+#include <cstring>
 
 using namespace std;
 
diff -Nur paintlib-2.6.2-orig/common/plbmpbase.cpp paintlib-2.6.2/common/plbmpbase.cpp
--- paintlib-2.6.2-orig/common/plbmpbase.cpp	2009-02-09 00:12:04.000000000 +0000
+++ paintlib-2.6.2/common/plbmpbase.cpp	2009-02-09 00:49:58.000000000 +0000
@@ -14,6 +14,7 @@
 #include "Filter/plfilterquantize.h"
 
 #include <iostream>
+#include <cstring>
 
 using namespace std;
 
diff -Nur paintlib-2.6.2-orig/common/plbmpdec.cpp paintlib-2.6.2/common/plbmpdec.cpp
--- paintlib-2.6.2-orig/common/plbmpdec.cpp	2009-02-09 00:12:04.000000000 +0000
+++ paintlib-2.6.2/common/plbmpdec.cpp	2009-02-09 00:29:40.000000000 +0000
@@ -15,6 +15,7 @@
 
 #include "plbmpdec.h"
 #include "plexcept.h"
+#include <cstring>
 
 PLBmpDecoder::PLBmpDecoder
 () : PLPicDecoder (),
diff -Nur paintlib-2.6.2-orig/common/plexif.cpp paintlib-2.6.2/common/plexif.cpp
--- paintlib-2.6.2-orig/common/plexif.cpp	2009-02-09 00:12:04.000000000 +0000
+++ paintlib-2.6.2/common/plexif.cpp	2009-02-09 00:39:21.000000000 +0000
@@ -27,6 +27,7 @@
 #include <cmath>
 
 #include <iostream> 
+#include <cstring>
 
 extern "C"
 {
diff -Nur paintlib-2.6.2-orig/common/plgifdec.cpp paintlib-2.6.2/common/plgifdec.cpp
--- paintlib-2.6.2-orig/common/plgifdec.cpp	2009-02-09 00:12:04.000000000 +0000
+++ paintlib-2.6.2/common/plgifdec.cpp	2009-02-09 01:02:33.000000000 +0000
@@ -15,6 +15,8 @@
 #include "planybmp.h"
 
 #include <stdio.h>
+#include <cstring>
+#include <string>
 
 extern "C"
 {
diff -Nur paintlib-2.6.2-orig/common/pljpegdec.cpp paintlib-2.6.2/common/pljpegdec.cpp
--- paintlib-2.6.2-orig/common/pljpegdec.cpp	2009-02-09 00:12:04.000000000 +0000
+++ paintlib-2.6.2/common/pljpegdec.cpp	2009-02-09 01:05:32.000000000 +0000
@@ -18,6 +18,8 @@
 #include "plexif.h"
 
 #include "jmemsrc.h"
+#include <string>
+#include <cstring>
 
 // This is for RGB_RED, RGB_GREEN, RGB_BLUE, RGB_PIXELSIZE
 #define JPEG_INTERNALS
diff -Nur paintlib-2.6.2-orig/common/pljpegenc.cpp paintlib-2.6.2/common/pljpegenc.cpp
--- paintlib-2.6.2-orig/common/pljpegenc.cpp	2009-02-09 00:12:04.000000000 +0000
+++ paintlib-2.6.2/common/pljpegenc.cpp	2009-02-09 00:58:55.000000000 +0000
@@ -31,6 +31,7 @@
 #include <jmorecfg.h>
 
 #include <stdio.h>
+#include <string.h>
 
 /////////////////////////////////////////////////////////////////////
 // Error handling.
diff -Nur paintlib-2.6.2-orig/common/plpcxdec.cpp paintlib-2.6.2/common/plpcxdec.cpp
--- paintlib-2.6.2-orig/common/plpcxdec.cpp	2009-02-09 00:12:04.000000000 +0000
+++ paintlib-2.6.2/common/plpcxdec.cpp	2009-02-09 00:35:41.000000000 +0000
@@ -13,6 +13,7 @@
 #include "plexcept.h"
 
 #include <iostream>
+#include <cstring>
 
 PLPCXDecoder::PLPCXDecoder()
   : PLPicDecoder()
diff -Nur paintlib-2.6.2-orig/common/plpgmdec.cpp paintlib-2.6.2/common/plpgmdec.cpp
--- paintlib-2.6.2-orig/common/plpgmdec.cpp	2009-02-09 00:12:04.000000000 +0000
+++ paintlib-2.6.2/common/plpgmdec.cpp	2009-02-09 00:31:12.000000000 +0000
@@ -13,6 +13,7 @@
 
 #include "plpgmdec.h"
 #include "plexcept.h"
+#include <cstring>
 
 PLPGMDecoder::PLPGMDecoder
     ()
diff -Nur paintlib-2.6.2-orig/common/plpicdec.cpp paintlib-2.6.2/common/plpicdec.cpp
--- paintlib-2.6.2-orig/common/plpicdec.cpp	2009-02-09 00:12:04.000000000 +0000
+++ paintlib-2.6.2/common/plpicdec.cpp	2009-02-09 00:32:36.000000000 +0000
@@ -27,6 +27,7 @@
 #include "planybmp.h"
 
 #include <stdio.h>
+#include <cstring>
 
 // Set default Trace configuration here. The defined levels are
 // explained in picdec.h.
diff -Nur paintlib-2.6.2-orig/common/plpicenc.cpp paintlib-2.6.2/common/plpicenc.cpp
--- paintlib-2.6.2-orig/common/plpicenc.cpp	2009-02-09 00:12:04.000000000 +0000
+++ paintlib-2.6.2/common/plpicenc.cpp	2009-02-09 00:26:48.000000000 +0000
@@ -19,6 +19,7 @@
 #include "plpicdec.h"
 
 #include <errno.h>
+#include <cstring>
 
 PLPicEncoder::PLPicEncoder()
   : PLObject()
diff -Nur paintlib-2.6.2-orig/common/plpictdec.cpp paintlib-2.6.2/common/plpictdec.cpp
--- paintlib-2.6.2-orig/common/plpictdec.cpp	2009-02-09 00:12:04.000000000 +0000
+++ paintlib-2.6.2/common/plpictdec.cpp	2009-02-09 00:23:47.000000000 +0000
@@ -25,6 +25,7 @@
 #include "plexcept.h"
 
 #include <stdio.h>
+#include <cstring>
 
 PLPictDecoder::PLPictDecoder
     ( PLJPEGDecoder * pJPEGDecoder
diff -Nur paintlib-2.6.2-orig/common/plpixelformat.cpp paintlib-2.6.2/common/plpixelformat.cpp
--- paintlib-2.6.2-orig/common/plpixelformat.cpp	2009-02-09 00:12:04.000000000 +0000
+++ paintlib-2.6.2/common/plpixelformat.cpp	2009-02-09 00:48:30.000000000 +0000
@@ -12,6 +12,8 @@
 
 #include <vector>
 #include <sstream>
+#include <cstring>
+#include <cstdlib>
 
 using namespace std;
 
diff -Nur paintlib-2.6.2-orig/common/plpsddec.cpp paintlib-2.6.2/common/plpsddec.cpp
--- paintlib-2.6.2-orig/common/plpsddec.cpp	2009-02-09 00:12:04.000000000 +0000
+++ paintlib-2.6.2/common/plpsddec.cpp	2009-02-09 00:42:12.000000000 +0000
@@ -15,6 +15,7 @@
 #include "plfilesrc.h"
 
 #include <stdio.h>
+#include <cstring>
 
 using namespace std;
 
diff -Nur paintlib-2.6.2-orig/common/plsgidec.cpp paintlib-2.6.2/common/plsgidec.cpp
--- paintlib-2.6.2-orig/common/plsgidec.cpp	2009-02-09 00:12:04.000000000 +0000
+++ paintlib-2.6.2/common/plsgidec.cpp	2009-02-09 00:44:22.000000000 +0000
@@ -11,6 +11,7 @@
 
 #include "plsgidec.h"
 #include "plexcept.h"
+#include <cstring>
 
 const PLWORD SGI_MAGIC = 474;
 
diff -Nur paintlib-2.6.2-orig/common/pltiffdec.cpp paintlib-2.6.2/common/pltiffdec.cpp
--- paintlib-2.6.2-orig/common/pltiffdec.cpp	2009-02-09 00:12:04.000000000 +0000
+++ paintlib-2.6.2/common/pltiffdec.cpp	2009-02-09 00:51:48.000000000 +0000
@@ -14,6 +14,7 @@
 #include "config.h"
 #include "pltiffdec.h"
 #include "plexcept.h"
+#include <cstring>
 
 extern "C"
 {
diff -Nur paintlib-2.6.2-orig/common/test/pltestdecoders.cpp paintlib-2.6.2/common/test/pltestdecoders.cpp
--- paintlib-2.6.2-orig/common/test/pltestdecoders.cpp	2009-02-09 00:12:04.000000000 +0000
+++ paintlib-2.6.2/common/test/pltestdecoders.cpp	2009-02-09 01:07:06.000000000 +0000
@@ -20,6 +20,7 @@
 
 #include <string>
 #include <iostream>
+#include <cstring>
 
 using namespace std;
 
